src/cudamatrix/cu-sparse-matrix.cc

```diff
@@ -70,7 +71,8 @@ template <typename Real>
 Real CuSparseMatrix<Real>::FrobeniusNorm() const {
 #if HAVE_CUDA == 1
   if (CuDevice::Instantiate().Enabled()) {
-    CuVector<Real> element_vec(*this);
+    CuVector<Real> element_vec(this->NumElements(), kUndefined);
+    this->CopyElementsToVec(&element_vec);
     return element_vec.Norm(2);
```

Comment:
This code is not right, you need account for the fact that the matrix's size is different from the number of nonzero elements.  Otherwise it's not really the frobenius norm.